src/wrap.c

```diff
@@ -2010,17 +2031,38 @@ inspectLib(struct dl_phdr_info *info, size_t size, void *data)
         uint64_t got_value = *((uint64_t *)got_addr);
         if (!fname || !got_value) continue;
 
+        char *file_from_got_value = osFileNameFromAddr(got_value);
+        if (!file_from_got_value) goto next;
+
+        // FYI: file_from_link_map is sometimes NULL and that's legit.
+        char *file_from_link_map = appview_realpath(info->dlpi_name, NULL);
+
         /*
          * Ignore got values that are resolved within it's own library.
          * These seem to be one of these two cases that aren't interesting:
          *   undefined symbols that haven't been called yet (resolve to plt section)
          *   locally defined external symbols (resolve to text section)
          */
-        char *file_from_maps_file = fileNameFromAddr(got_value);
-        char *file_from_link_map = appview_realpath(info->dlpi_name, NULL);
-        int found_locally = file_from_link_map && !appview_strcmp(file_from_maps_file, file_from_link_map);
+        int found_locally = file_from_link_map && !appview_strcmp(file_from_got_value, file_from_link_map);
         if (found_locally) goto next;
 
+        if ((file_from_link_map && !appview_strstr(file_from_link_map, "libappview")) &&
```

Comment:
no need to check for file_from_link_map again here

Reply:
i think we do need to check because in the previous check we require the null check AND the string compare in order to goto next.

file_from_link_map && !appview_strcmp(file_from_got_value, file_from_link_map);

Reply:
you're right